### PR TITLE
[prometheus] add port name

### LINKED ISSUE #3742: [prometheus] Deployment and Statefulset: add container port name
### Is your feature request related to a problem ?

In some cases it is necessary to be able to set the port name of the prometheus container

### Describe the solution you'd like.

The simplest, you need to add a parameter describing the name of the container
```
server:
  portName: web
```
```
...
ports:
  - containerPort: 9090
{{- if .Values.server.portName }}
    name: {{ .Values.server.portName }}
{{- end }}
...
```

### Describe alternatives you've considered.

Have to make patches

### Additional context.

_No response_

### PR DESCRIPTION
**Pull request by DrFaust92**
#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #3742

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus/templates/deploy.yaml [hunk 1]
```diff
@@ -147,6 +147,9 @@ spec:
         {{- end }}
           ports:
             - containerPort: 9090
+          {{- if .Values.server.portName }}
+              portName: {{ .Values.server.portName }}
```
Comment (zeritti) [inline charts/prometheus/templates/deploy.yaml:151]:
```suggestion
              name: {{ .Values.server.portName }}
```

File: charts/prometheus/templates/sts.yaml [hunk 2]
```diff
@@ -144,6 +144,9 @@ spec:
         {{- end }}
           ports:
             - containerPort: 9090
+          {{- if .Values.server.portName }}
+              portName: {{ .Values.server.portName }}
```
Comment (zeritti) [inline charts/prometheus/templates/sts.yaml:148]:
```suggestion
              name: {{ .Values.server.portName }}
```

The new field should be present in the values file.

**Review by zeritti — CHANGES_REQUESTED** [review]

**Comment by DrFaust92** [comment]
Thanks [zeritti](https://github.com/zeritti), added.

File: charts/prometheus/values.yaml [hunk 3]
```diff
@@ -565,6 +565,9 @@ server:
   # Use hostPort
   # hostPort: 9090
 
+  # Use portName
+  # portName: prometheus
```
Comment (zeritti) [inline charts/prometheus/values.yaml:569]: Please, do not comment out fields.

```suggestion
  portName: ""
```

Comment (DrFaust92) [reply charts/prometheus/values.yaml]: 👍

**Review by zanhsieh — APPROVED** [review]

**Review by zeritti — APPROVED** [review]
LGTM

**Pull request merged**